src/khepri.erl

```diff
@@ -133,69 +142,56 @@
                            compare_and_swap/3, compare_and_swap/4,
                            exists/2,
                            has_data/2,
-                           get_data/2,
-                           get_data_or/3,
-                           has_sproc/2,
+                           is_sproc/2,
                            run_sproc/3,
-                           transaction/2, transaction/3]}).
+                           transaction/2, transaction/3,
+
+                           unwrap_result/1]}).
 
 -type store_id() :: atom().
 %% ID of a Khepri store.
 %%
 %% This is the same as the Ra cluster name hosting the Khepri store.
 
--type error(Type) :: {error, Type}.
-%% Return value of a failed command or query.
-
 -type data() :: any().
-%% Data stored in a node's payload.
+%% Data stored in a tree node's payload.
 
 -type payload_version() :: pos_integer().
-%% Number of changes made to the payload of a node.
+%% Number of changes made to the payload of a tree node.
 %%
-%% The payload version starts at 1 when a node is created. It is increased by 1
-%% each time the payload is added, modified or removed.
+%% The payload version starts at 1 when a tree node is created. It is increased
+%% by 1 each time the payload is added, modified or removed.
 
 -type child_list_version() :: pos_integer().
-%% Number of changes made to the list of child nodes of a node (child nodes
-%% added or removed).
+%% Number of changes made to the list of child nodes of a tree node (child
+%% nodes added or removed).
 %%
-%% The child list version starts at 1 when a node is created. It is increased
-%% by 1 each time a child is added or removed. Changes made to existing nodes
-%% are not reflected in this version.
+%% The child list version starts at 1 when a tree node is created. It is
+%% increased by 1 each time a child is added or removed. Changes made to
+%% existing nodes are not reflected in this version.
 
 -type child_list_length() :: non_neg_integer().
 %% Number of direct child nodes under a tree node.
 
 -type node_props() ::
-    #{data => data(),
+    #{data => khepri:data(),
+      has_data => boolean(),
       sproc => khepri_fun:standalone_fun(),
-      payload_version => payload_version(),
-      child_list_version => child_list_version(),
-      child_list_length => child_list_length(),
-      child_nodes => #{khepri_path:node_id() => node_props()}}.
+      is_sproc => boolean(),
```

Comment:
I find another benefit for `data`/`sproc`: the spec is more precise and Dialyzer will be more helpful. If we have `payload => any() | khepri_fun:standalone_fun()`, Dialyzer won't be able to detect any errors.

But I agree that the inconsistency between the `props_to_return` names and the actually returned properties names is unfortunate.

I'm slightly leaning towards keeping `data` and `sproc` because of the benefits with Dialyzer, especially considering that the option will probably be rarely used.

Reply:
I don't feel strongly about putting them together. Let's keep `data` and `sproc` 👍